src/retention-management/nuxeo-retention-technical-overview.md

```diff
@@ -0,0 +1,166 @@
+    status: 'ok'
+labels:
+    - lts2019-wip
+    - jaubenque
+    - mlumeau
+    - retention-management
+toc: true
+tree_item_index: 100
+private: true
+---
+
+## Amazon S3 Compliance mode integration
+
+### {{> anchor 's3-configuration-requirements'}} Amazon S3 configuration requirements
+
+Compliance with SEC Rule 17a-4 requires to configure Nuxeo Platform to utilize Amazon S3 Compliance Buckets to store books and records required by regulation, thereby establishing the foundation for meeting the requirements of the Rule. Specifically,
+
+- Direct writes to the Amazon S3 storage system must be disabled, to ensure all documents transit through Nuxeo Platform for compliant processing.
+
+- The Amazon S3 Object Lock feature must be enabled in Compliance Mode on the bucket intended to store objects that are marked as final records.
+
+- Amazon S3 Versioning must be enabled.
+
+- The default retention value for Amazon S3 Compliance Buckets intended to retain compliant record objects must be set to zero (0).
+
+- No Min/Max range should be established for Amazon S3 Compliance Buckets intended to retain compliant record objects.
+
+- Amazon S3 Lifecycle Policies must not be configured for use within the Nuxeo Platform storage subsystem.
+
+
+### Amazon S3 buckets
+
+Nuxeo Platform with Nuxeo Retention Management add-on requires the usage of 2 Amazon buckets:
+
+- A standard S3 bucket as for any other standard Nuxeo instance,
+
+  - This bucket is used to store the standard documents,
+
+
+- A S3 bucket dedicated to the records that is configured on [Compliance mode]({{page page='nuxeo-retention-technical-overview'}}#s3-configuration-requirements):,
+
+  - This bucket is used to store the records only,
+  - Compared to the standard bucket, the Garbage Collector must be disabled as there is a [specific deletion process]({{page page='nuxeo-retention-technical-overview'}}#record-deletion-flow),
+
+
+## {{> anchor 'dedicated-low-fields'}} Dedicated low-level fields
+
+In order to provide required data to Amazon S3 (expiration date and legal hold) and to ensure that records can’t be deleted when not eligible, some new low-level fields have been added at the storage level:
+
+- **ecm:isRecord** (boolean) - True if the document is declared as a record
+- **ecm:retainUntil** (datetime) - Expiration date (end of retention period)
+- **ecm:hasLegalHold** (boolean) - True if there is a legal hold applied to the record
+- **ecm:isUndeletable** (boolean) - True if the record is NOT eligible to deletion
+
+If a legal hold is applied to a document or if a retention rule is applied to a document, then:
+
+- modification of the main blob (file:content) is prevented
+
+  - this includes modification through restoring an older version,
+
+
+- deletion (including recursive delete) is prevented:
+
+  - this is enforced at low-level,
+  - this information is made available per-document through a security policy,
+
+
+- ecm:isUndeletable is set to true,
```

Comment:
This doesn't exist anymore but there's a `ecm:isUnderRetentionOrLegalHold` synthetic property exported in JSON and corresponding method for the caller's convenience.